clippy_lints/src/non_copy_const.rs

```diff
@@ -34,6 +34,14 @@ declare_clippy_lint! {
     /// `std::sync::ONCE_INIT` constant). In this case the use of `const` is legit,
     /// and this lint should be suppressed.
     ///
+    /// When an enum has variants with interior mutability, use of its non interior mutable
+    /// variants can generate false positives. See issue [#3962](https://github.com/rust-lang/rust-clippy/issues/3962)
+    ///
+    /// Types that have underlying or potential interior mutability trigger the lint, but are
+    /// usually not the type of errors that the lint is trying to catch. See issues
```

Comment:
Well, that is exactly the issue this lint is trying to catch. But for some times the interior mutability is not a problem.

Reply:
Yeah this is why I thought that the lint was right in the end. The lint checks the type, not the actual use. For the enum case it's wrong because the interior mutable field can't be accessed, but if it's not an enum, there is potential for that field to be accessed (and if it's not used, why is it there?).

But I agree that we should specify this in the doc: the lint will trigger even if the interior mutable field is not used.